compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -138,6 +138,9 @@ impl<'psess, 'src> StringReader<'psess, 'src> {
             if let rustc_lexer::TokenKind::Semi
             | rustc_lexer::TokenKind::LineComment { .. }
             | rustc_lexer::TokenKind::BlockComment { .. }
+            | rustc_lexer::TokenKind::OpenParen
+            | rustc_lexer::TokenKind::OpenBrace
+            | rustc_lexer::TokenKind::OpenBracket
```

Comment:
I'm keeping this change because we still use `last_lifetime` for string literals (this will make suggestions for `'foo (bar)'" that was meant to be a string to stop working, but that should be fine.

Reply:
Why is this beneficial? The only time I could see us not want to make string lit suggestions is:
1. If the string literal has a newline in it
2. The number of delimiters are inbalanced between the start and end of the string 


But the presence of delims don't really make it any less likely you meant a string lit imo

Reply:
Beneficial in the sense of being more conservative, but I'm not strongly inclined either way. Removed the change.